tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -29,18 +54,36 @@ func (r *RealDependencies) execute(cmd string, redirectStdout bool, args ...stri
 	fmt.Printf("%s %s\n", cmd, strings.Join(args, " "))
 	externalCommand := exec.Command(cmd, args...)
 	externalCommand.Stdout = os.Stdout
+
+	var stderr bytes.Buffer
 	// TODO Check naming and redirection logic
 	if !redirectStdout {
-		externalCommand.Stderr = os.Stderr
+		externalCommand.Stderr = &stderr
 	}
-	return externalCommand.Run()
+
+	err := externalCommand.Run()
+	stderrStr := stderr.String()
+	if err != nil && !redirectStdout {
+		// Print stderr to os.Stderr.
+		fmt.Println(stderrStr)
+
+		// Parse the exit status code.
+		exitcode, _ := strconv.Atoi(strings.Trim(strings.SplitN(err.Error(), "exit status", 2)[1], " "))
```

Comment:
Use the standard library to get the [ExitCode](https://golang.org/pkg/os/#ProcessState.ExitCode).
This logic is too fragile and unnecessary.

Reply:
Thanks, updated by leveraging the standard lib.